src/reconcile/rec_write.c

```diff
@@ -2649,7 +2653,7 @@ __rec_split_fixup(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 	 * WT_PAGE_HEADER header onto the scratch buffer, most of the header
 	 * information remains unchanged between the pages.
 	 */
-	WT_RET(__wt_scr_alloc(session, r->page_size, &tmp));
+	WT_RET(__wt_scr_alloc(session, r->dsk.memsize, &tmp));
```

Comment:
@keithbostic I'm not entirely comfortable that r->dsk.memsize is the right thing to use here - though other callers of ```__rec_split_write``` pass ```r->dsk``` directly, so I guess it's the safe solution.